morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -256,6 +262,38 @@ public void ensureCaps() {
     }
 
     @Override
+    public void enableDocumentValidation() {
+        for (final MappedClass mc : mapper.getMappedClasses()) {
+            process(mc, (Validation) mc.getAnnotation(Validation.class));
+        }
+    }
+
+    void process(final MappedClass mc, final Validation validation) {
+        if (validation != null) {
+            String collectionName = mc.getCollectionName();
+            CommandResult result = getDB()
+                .command(new BasicDBObject("collMod", collectionName)
+                             .append("validator", parse(validation.value()))
+                             .append("validationLevel", validation.level().getValue())
+                             .append("validationAction", validation.action().getValue())
+                        );
+
+            if (!result.ok()) {
+                if (result.getInt("code") == 26) {
+                    ValidationOptions options = new ValidationOptions()
+                        .validator(parse(validation.value()))
+                        .validationLevel(validation.level())
+                        .validationAction(validation.action());
+                    getDatabase().createCollection(collectionName, new CreateCollectionOptions().validationOptions(options));
+                } else {
+                    throw new MappingException(format("Could not add document validation on '%s:'  %s", collectionName,
```

Comment:
Not sure what you do in other places, but result.throwOnError() will throw an exception that includes pertinent information like the error code.